### ISSUE TITLE
Support SPARQL endpoints with multiple SPARQL service description

### ISSUE THREAD

**Issue by marcelomachado** [comunica/comunica#1061]
#### Issue type:

- :bug: Bug 

____
#### Description:

I created a small web app that queries multiple RDF sources, including a Blazegraph SPARQL endpoint. If my only Comunica source is this Blazegraph SPARQL endpoint, I get the expected result. When I add another source and execute the query, it returns a 404 ERROR:
```
Invalid SPARQL endpoint response from http://{host}/bigdata/LBS/namespace/{namespace}/sparql (HTTP status 404).
```

However, I realized that the endpoint that appears in the error message is different than the one I was trying to use: 
```
http://{host}/blazegraph/namespace/{namespace}/sparql
```

That's when I decided to look at the service description and realized that there are multiple endpoints and in fact the one in the error message is not found:

```
<endpoint xmlns="http://www.w3.org/ns/sparql-service-description#" rdf:resource="http://{host}/blazegraph/namespace/{namespace}/sparql"/>
<endpoint xmlns="http://www.w3.org/ns/sparql-service-description#" rdf:resource="http://{host}/bigdata/LBS/namespace/{namespace}/sparql"/>
```

I don't have access to the Blazegraph service to remove such a description, actually I believe this shouldn't be the solution. I can't see why the query couldn't be completed with the endpoint that works (the endpoint I'm using in `sources`). I also don't understand why this is a problem that only happens when more than one source is selected.

FYI: This issue https://github.com/comunica/comunica/issues/535 seems to be similar, however the problem occurs when no endpoint is provided in the service description.

**Comment by rubensworks** [comment]
I have no experience with Blazegraph myself, but I would guess this is a Blazegraph configuration issue.
If `http://{host}/bigdata/LBS/namespace/{namespace}/sparql` is exposed in the SD, then this would mean that it is enabled in Blazegraph, but is somehow broken.

As a workaround for this Blazegraph bug, it _may_ be possible to [force the `sparql` source type](https://comunica.dev/docs/query/advanced/source_types/).

**Comment by rubensworks** [comment]
Hmm, too bad the workaround doesn't work.

The proper long-term solution would be to consider the two endpoints in the SD as alternatives to the same dataset. In this case, Comunica should first attempt to query the first one, and fallback to the second one if the first one fails (or some other load-balancing solution). However, implementing this would require quite some effort.

A short-term solution could be to add a context option in the SPARQL endpoint actor that disables/ignores the endpoints detected in the SD, which would require changes here: https://github.com/comunica/comunica/blob/master/packages/actor-rdf-resolve-hypermedia-sparql/lib/ActorRdfResolveHypermediaSparql.ts#L30

Other than these options, I see no solutions that require no code changes.

PRs or bounties on this are welcome!

**Comment by marcelomachado** [comment]
I am biased to the first approach you described and the start point would be make the `sparqlEndpoint` field an array [here](https://github.com/comunica/comunica/blob/efa0e9192f0a22d242744779f0286228de92959f/packages/actor-rdf-metadata-extract-sparql-service/lib/ActorRdfMetadataExtractSparqlService.ts#L31). 

However, before going too deep into this solution I will first submit a more conservative PR considering the second approach you mentioned. When I'm more familiar with the architecture and code, I'll go back to the first option.

**Comment by rubensworks** [comment]
> I am biased to the first approach you described and the start point would be make the sparqlEndpoint field an array [here](https://github.com/comunica/comunica/blob/efa0e9192f0a22d242744779f0286228de92959f/packages/actor-rdf-metadata-extract-sparql-service/lib/ActorRdfMetadataExtractSparqlService.ts#L31).

Indeed! We'll also need a datastructure change to provide alternatives for datasources, and a mechanism to switch to an alternative. Nothing too complicated, but may need changes in quite a few places.

> However, before going too deep into this solution I will first submit a more conservative PR considering the second approach you mentioned. When I'm more familiar with the architecture and code, I'll go back to the first option.

Sounds great! Just give me a shout if you need any input.

**Comment by rubensworks** [comment]
Also related to #980
